src/effects/VST3/VST3Utils.cpp

```diff
@@ -73,3 +76,70 @@ bool VST3Utils::ParseAutomationParameterKey(const wxString& key, Steinberg::Vst:
    return false;
 
 }
+
+
+
+Steinberg::IPtr<PresetsBufferStream> PresetsBufferStream::fromString(const wxString& str)
+{
+   Steinberg::Buffer buffer(str.size() / 4 * 3);
+   auto len = Base64::Decode(str, buffer);
+   assert(len <= buffer.getSize());
+   buffer.setSize(len);
+
+   auto result = owned(safenew PresetsBufferStream);
+   result->mBuffer.take(buffer);
+   return result;
+}
+
+
+wxString PresetsBufferStream::toString() const
+{
+   return Base64::Encode(mBuffer, mBuffer.getSize());
+}
+
+
+
+bool VST3Wrapper::FetchSettings(VST3EffectSettings& vst3Settings) const
```

Comment:
`FetchSettings` is given settings that are not assumed to be initially empty.  Be sure to assign nullopt along paths that fail to get state from the plug-in.

Reply:
ok, done